cas-mfa-java/src/main/java/net/unicon/cas/mfa/authentication/RegexAuthenticationMethodTranslator.java

```diff
@@ -0,0 +1,60 @@
+package net.unicon.cas.mfa.authentication;
+
+import net.unicon.cas.mfa.web.support.UnrecognizedAuthenticationMethodException;
+import org.jasig.cas.authentication.principal.WebApplicationService;
+
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * A translator that will check a list of regex patterns and return an authentication method name.
+ *
+ * @author John Gasper
+ */
+public class RegexAuthenticationMethodTranslator implements AuthenticationMethodTranslator {
+    private final Map<Pattern, String> translationMap;
+
+    private boolean ignoreIfNoMatchIsFound = true;
+
+    private String defaultMfaMethod = "";
+
+    /**
+     * Instantiates a new Regex authentication method translator.
+     *
+     * @param translationMap the regex/mfa method translation map
+     */
+    public RegexAuthenticationMethodTranslator(final Map<String, String> translationMap) {
+        final Map<Pattern, String> optimizedMap = new HashMap<Pattern, String>();
+
+        for (final String pattern : translationMap.keySet()) {
+            optimizedMap.put(Pattern.compile(pattern), translationMap.get(pattern));
+        }
+
+        this.translationMap = optimizedMap;
+    }
+
+    public void setIgnoreIfNoMatchIsFound(final boolean ignoreIfNoMatchIsFound) {
```

Comment:
This really is unnecessary. Use the defaultMfaMethod. If null, throw exceptions, otherwise return that one.